src/main/java/ru/ralsei/whatcanyousee/gameactivity/InternetConnector.java

```diff
@@ -0,0 +1,267 @@
+package ru.ralsei.whatcanyousee.gameactivity;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.google.android.gms.games.multiplayer.Multiplayer;
+import com.google.android.gms.games.multiplayer.realtime.OnRealTimeMessageReceivedListener;
+import com.google.android.gms.games.multiplayer.realtime.RealTimeMessage;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+/**
+ * Class for managing internet messaging between player's.
+ */
+class InternetConnector {
+    private String TAG = "What can you see: Internet connector";
+
+    private GameActivity activity;
+
+    InternetConnector(GameActivity activity) {
+        this.activity = activity;
+    }
+
+    /**
+     * True if got message from teammate that he is ready to get a voice connection.
+     */
+    private boolean otherPlayerIsReady = false;
+
+    /**
+     * True if ready to send and receive voice.
+     */
+    private boolean prepared = false;
+
+    /**
+     * Reset prepared status for the next created game.
+     */
+    void clearResources() {
+        otherPlayerIsReady = false;
+        prepared = false;
+    }
+
+    /**
+     * Send to other player that we are ready to receive his voice.
+     * Also sends game settings if we are the game host.
+     */
+    void sendReadyMessage() {
+        byte[] message;
+
+        GameplayHandler.GameSettings gameSettings = activity.getGameplayHandler().getGameSettings();
+        if (gameSettings != null) {
+            try {
+                ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+                ObjectOutputStream writeStream;
+
+                writeStream = new ObjectOutputStream(byteStream);
+
+                gameSettings.flipSettings();
+                writeStream.writeObject(gameSettings);
+                writeStream.flush();
+                gameSettings.flipSettings();
+
+                message = byteStream.toByteArray();
+                writeStream.close();
```

Comment:
If we don't reach this point, the writer will never be closed (i.e. will leak). Use can use this auto-close feature in Java to simplify such code:

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Same elsewhere.